PR-Review-Guide.md

```diff
@@ -0,0 +1,129 @@
+* [High level considerations](#high-level-considerations)
+    * [Security](#security)
+    * [Reliability](#reliability)
+    * [Performance](#performance)
+    * [Maintenance](#maintenance)
+    * [Usability](#usability)
+* [Specifics to consider](#specifics-to-consider)
+    * [Basics](#basics)
+    * [Layout and formatting](#layout-and-formatting)
+    * [Design](#design)
+    * [Code comments](#code-comments)
+    * [Documentation](#documentation)
+    * [Logging and Debugging](#logging-and-debugging)
+    * [Testing](#testing)
+    * [Environments](#environments)
+    * [Upgrades](#upgrades)
+* [Mandatory Checks](#mandatory-checks)
+
+---
+
+This document attempts to capture some of the points to consider when reviewing a [new code contribution](CONTRIBUTING.md) (a PR) to ensure overall quality is kept as high as possible.
+
+## High level considerations
+
+### Security
+
+- Could the change introduce a potential security risk?
+  - What additional checks would resolve it?
+
+### Reliability
+
+- What happens on failure? (Consider **every** possible failure)
+- If the code crashes, what is the overall system impact?
+
+### Performance
+
+- Could this change negatively impact performance?
+- Can any calls block? If so, there should be a log call before the blocking call explaining what is about to be done.
+
+### Maintenance
+
+- How easy would this code be to maintain:
+  - By an experienced developer on the project?
+  - By a developer with none / almost no experience of the code base or the project?
+
+### Usability
+
+- Does the change improve or worsen usability?
+- Could usability be improved further?
+
+## Specifics to consider
+
+### Basics
+
+- Consider *all* inputs and outputs.
+- What resources are being used? (memory, disk, *etc*) Are they released?
+- Are all return values and arguments checked?
+- Are there any magic numbers?
+
+### Layout and formatting
+
+- Is formatting and naming consistent?
+- Are there any spelling mistakes ("typos")?
+- Do all new files contain the required licence header?
+
+### Design
+
+- Could the code be simplified / refactored?
+  - Is the code "too clever"? Overly clever code tends to be fragile: reject it unless there is a **very** good documented reason!
+- Is there a better (simpler) solution to the problem?
+- Is there any duplication?
+- Is the code "re-inventing the wheel?" Is there a standard package that could be used instead?
+- What assumptions does the code make? Are they valid? Are they documented?
+- Does the code make sense?
+  - Could someone with minimal exposure to the codebase guess at what it is doing? (hint: they should be able to).
+
+### Code comments
+
+- Is the code sufficiently well commented?
+- Are all functions and function parameters documented (or for `golang` programs, is they purpose obvious?)
```

Comment:
"is **they** purpose obvious?" should be "is **their** purpose obvious?"

Reply:
Good catch! Fixed.